### DISCUSSION TITLE
The recipe produces a Postgresql VALUES construct doesn't work on version 1.4.42

### DISCUSSION THREAD

**Discussion by tomaszkingukrol** [sqlalchemy/sqlalchemy#8655] [Usage Questions] [answered]
### Describe the bug
Recipe described here: https://github.com/sqlalchemy/sqlalchemy/wiki/PGValues doesn't work on SQLAlchemy 1.4.42.
### To Reproduce
https://github.com/sqlalchemy/sqlalchemy/wiki/PGValues
### Error
`Traceback (most recent call last):
  File ".../site-packages/sqlalchemy/sql/base.py", line 1223, in __getattr__
    return self._index[key]
KeyError: 'mykey'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):

  File ".../site-packages/sqlalchemy/sql/base.py", line 1225, in __getattr__
    util.raise_(AttributeError(key), replace_context=err)
  File ".../site-packages/sqlalchemy/util/compat.py", line 208, in raise_
    raise exception
AttributeError: mykey`

### Versions

- OS: mac os 12.2.1
- Python: 3.10.5
- SQLAlchemy: 1.4.42
- Database: Postgresql
- DBAPI (psycopg)

**Comment by zzzeek** [comment, marked as answer]
Hey there -

this recipe is very old, SQLAlchemy includes a [values](https://docs.sqlalchemy.org/en/14/core/selectable.html?highlight=values#sqlalchemy.sql.expression.values) construct built in.    I will note this on the wiki.

**Reply by tomaszkingukrol** [reply]
Thank you for your answer. It works!
